00_setup.Rmd

```diff
@@ -395,143 +304,147 @@ get_xg3_temp <- function(locs,
   
   
   DBI::dbDisconnect(watina)
-  
+  DBI::dbDisconnect(watina2)
   #volgende watina-functie werkt ook zonder databankconnectie, maar omdat deze op sommige pc's niet stabiel werkt, worden de data hier alvast lokaal weggeschreven.
     
-filter_xg3_lok <- function(data,
-                           xg3_type) {
-  
-  assertthat::assert_that(all(c("loc_code", "hydroyear") %in% colnames(data)) &
-                any(grepl("g3", colnames(data))),
-              msg = "data does not have the necessary 'loc_code', 'hydroyear' and XG3 columns.")
-  
-  assertthat::assert_that(("L" %in% xg3_type & any(grepl("lg3", colnames(data)))) |
-                ("H" %in% xg3_type & any(grepl("hg3", colnames(data)))) |
-                ("V" %in% xg3_type & any(grepl("vg3", colnames(data)))),
-              msg = paste("xg3_type is set to",
-                          dput(xg3_type),
-                          "but at least one XG3 type is missing in data.")
-  )
-  
-  xg3_filtered <-
-    data %>%
-    select(.data$loc_code,
-           .data$hydroyear,
-           if ("L" %in% xg3_type) contains("lg"),
-           if ("H" %in% xg3_type) contains("hg"),
-           if ("V" %in% xg3_type) contains("vg")
-    ) %>%
-    arrange(.data$loc_code,
-            .data$hydroyear)
-  
-  return(xg3_filtered)
-}
-
-eval_xg3_series_lok <- function(data,
-                                xg3_type = c("L", "H", "V"),
-                                max_gap,
-                                min_dur){
-  if (missing(xg3_type)) {
-    xg3_type <- match.arg(xg3_type)} else {
-      assertthat::assert_that(all(xg3_type %in%
-                        c("L", "H", "V")),
-                  msg = "You specified at least one unknown xg3_type.")
+  filter_xg3_lok <- function(data,
```

Comment:
If this and below functions, or parts of them, are interesting to share with others (I didn't go into them), best consider making a pull request in the watina repo.

Reply:
I needed the use these watina-functions locally, because otherwise i became some errors eg. 'cannot open file 'Boundary_Crossing_Time.txt': Permission denied'
The content of the functions remained the same (<Watina version 0.3.0)

Reply:
I see. It seems to have to do with the KSgeneral package. watina calls `KSgeneral::disc_ks_test()`, which calls `KSgeneral::ks_c_cdf_Rcpp()`. For the latter to work, `disc_ks_test()` [writes](https://github.com/raymondtsr/ksgeneral/blob/master/R/disc_ks_test.R#L182) a file "Boundary_Crossing_Time.txt". It  appears this doesn't work on your PC.

Now I remember we already met this - #6.

Reply:
Yes indeed, that's the problem (on my laptop)